// ngx-pipes/src/app/pipes/string/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var ucwords_1 = require("./ucwords");
var ltrim_1 = require("./ltrim");
var repeat_1 = require("./repeat");
var rtrim_1 = require("./rtrim");
var scan_1 = require("./scan");
var shorten_1 = require("./shorten");
var strip_tags_1 = require("./strip-tags");
var trim_1 = require("./trim");
var ucfirst_1 = require("./ucfirst");
var slugify_1 = require("./slugify");
var camelize_1 = require("./camelize");
var latinise_1 = require("./latinise");
var lines_1 = require("./lines");
var underscore_1 = require("./underscore");
var match_1 = require("./match");
var test_1 = require("./test");
var lpad_1 = require("./lpad");
var rpad_1 = require("./rpad");
exports.STRING_PIPES = [
    ltrim_1.LeftTrimPipe, repeat_1.RepeatPipe, rtrim_1.RightTrimPipe, scan_1.ScanPipe, shorten_1.ShortenPipe,
    strip_tags_1.StripTagsPipe, trim_1.TrimPipe, ucfirst_1.UcFirstPipe, ucwords_1.UcWordsPipe, slugify_1.SlugifyPipe,
    camelize_1.CamelizePipe, latinise_1.LatinisePipe, lines_1.LinesPipe, underscore_1.UnderscorePipe, match_1.MatchPipe,
    test_1.TestPipe, lpad_1.LeftPadPipe, rpad_1.RightPadPipe
];
var NgStringPipesModule = (function () {
    function NgStringPipesModule() {
    }
    return NgStringPipesModule;
}());
NgStringPipesModule.decorators = [
    { type: core_1.NgModule, args: [{
                declarations: exports.STRING_PIPES,
                imports: [],
                exports: exports.STRING_PIPES
            },] },
];
/** @nocollapse */
NgStringPipesModule.ctorParameters = function () { return []; };
exports.NgStringPipesModule = NgStringPipesModule;
var ucwords_2 = require("./ucwords");
exports.UcWordsPipe = ucwords_2.UcWordsPipe;
var ltrim_2 = require("./ltrim");
exports.LeftTrimPipe = ltrim_2.LeftTrimPipe;
var repeat_2 = require("./repeat");
exports.RepeatPipe = repeat_2.RepeatPipe;
var rtrim_2 = require("./rtrim");
exports.RightTrimPipe = rtrim_2.RightTrimPipe;
var scan_2 = require("./scan");
exports.ScanPipe = scan_2.ScanPipe;
var shorten_2 = require("./shorten");
exports.ShortenPipe = shorten_2.ShortenPipe;
var strip_tags_2 = require("./strip-tags");
exports.StripTagsPipe = strip_tags_2.StripTagsPipe;
var trim_2 = require("./trim");
exports.TrimPipe = trim_2.TrimPipe;
var ucfirst_2 = require("./ucfirst");
exports.UcFirstPipe = ucfirst_2.UcFirstPipe;
var slugify_2 = require("./slugify");
exports.SlugifyPipe = slugify_2.SlugifyPipe;
var camelize_2 = require("./camelize");
exports.CamelizePipe = camelize_2.CamelizePipe;
var latinise_2 = require("./latinise");
exports.LatinisePipe = latinise_2.LatinisePipe;
var lines_2 = require("./lines");
exports.LinesPipe = lines_2.LinesPipe;
var underscore_2 = require("./underscore");
exports.UnderscorePipe = underscore_2.UnderscorePipe;
var match_2 = require("./match");
exports.MatchPipe = match_2.MatchPipe;
var test_2 = require("./test");
exports.TestPipe = test_2.TestPipe;
var lpad_2 = require("./lpad");
exports.LeftPadPipe = lpad_2.LeftPadPipe;
var rpad_2 = require("./rpad");
exports.RightPadPipe = rpad_2.RightPadPipe;
