// ngx-pipes/src/app/pipes/object/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var keys_1 = require("./keys");
var values_1 = require("./values");
var pairs_1 = require("./pairs");
var pick_1 = require("./pick");
var omit_1 = require("./omit");
var invert_1 = require("./invert");
var invert_by_1 = require("./invert-by");
var diff_obj_1 = require("./diff-obj");
var core_1 = require("@angular/core");
var OBJECT_PIPES = [
    keys_1.KeysPipe, values_1.ValuesPipe, pairs_1.PairsPipe, pick_1.PickPipe, invert_1.InvertPipe, invert_by_1.InvertByPipe,
    omit_1.OmitPipe, diff_obj_1.DiffObjPipe
];
var NgObjectPipesModule = (function () {
    function NgObjectPipesModule() {
    }
    return NgObjectPipesModule;
}());
NgObjectPipesModule.decorators = [
    { type: core_1.NgModule, args: [{
                declarations: OBJECT_PIPES,
                imports: [],
                exports: OBJECT_PIPES
            },] },
];
/** @nocollapse */
NgObjectPipesModule.ctorParameters = function () { return []; };
exports.NgObjectPipesModule = NgObjectPipesModule;
var keys_2 = require("./keys");
exports.KeysPipe = keys_2.KeysPipe;
var values_2 = require("./values");
exports.ValuesPipe = values_2.ValuesPipe;
var pairs_2 = require("./pairs");
exports.PairsPipe = pairs_2.PairsPipe;
var pick_2 = require("./pick");
exports.PickPipe = pick_2.PickPipe;
var omit_2 = require("./omit");
exports.OmitPipe = omit_2.OmitPipe;
var invert_2 = require("./invert");
exports.InvertPipe = invert_2.InvertPipe;
var invert_by_2 = require("./invert-by");
exports.InvertByPipe = invert_by_2.InvertByPipe;
var diff_obj_2 = require("./diff-obj");
exports.DiffObjPipe = diff_obj_2.DiffObjPipe;
