// ngx-pipes/src/app/pipes/array/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var diff_1 = require("./diff");
var initial_1 = require("./initial");
var flatten_1 = require("./flatten");
var intersection_1 = require("./intersection");
var reverse_1 = require("./reverse");
var tail_1 = require("./tail");
var truthify_1 = require("./truthify");
var union_1 = require("./union");
var unique_1 = require("./unique");
var without_1 = require("./without");
var pluck_1 = require("./pluck");
var shuffle_1 = require("./shuffle");
var every_1 = require("./every");
var some_1 = require("./some");
var sample_1 = require("./sample");
var group_by_1 = require("./group-by");
var filter_by_1 = require("./filter-by");
var order_by_1 = require("./order-by");
var core_1 = require("@angular/core");
var ARRAY_PIPES = [
    diff_1.DiffPipe, flatten_1.FlattenPipe, initial_1.InitialPipe, intersection_1.IntersectionPipe, reverse_1.ReversePipe, tail_1.TailPipe,
    truthify_1.TrurthifyPipe, union_1.UnionPipe, unique_1.UniquePipe, without_1.WithoutPipe, pluck_1.PluckPipe, shuffle_1.ShufflePipe,
    every_1.EveryPipe, some_1.SomePipe, sample_1.SamplePipe, group_by_1.GroupByPipe, filter_by_1.FilterByPipe, order_by_1.OrderByPipe
];
var NgArrayPipesModule = (function () {
    function NgArrayPipesModule() {
    }
    return NgArrayPipesModule;
}());
NgArrayPipesModule.decorators = [
    { type: core_1.NgModule, args: [{
                declarations: ARRAY_PIPES,
                imports: [],
                exports: ARRAY_PIPES
            },] },
];
/** @nocollapse */
NgArrayPipesModule.ctorParameters = function () { return []; };
exports.NgArrayPipesModule = NgArrayPipesModule;
var diff_2 = require("./diff");
exports.DiffPipe = diff_2.DiffPipe;
var initial_2 = require("./initial");
exports.InitialPipe = initial_2.InitialPipe;
var flatten_2 = require("./flatten");
exports.FlattenPipe = flatten_2.FlattenPipe;
var intersection_2 = require("./intersection");
exports.IntersectionPipe = intersection_2.IntersectionPipe;
var reverse_2 = require("./reverse");
exports.ReversePipe = reverse_2.ReversePipe;
var tail_2 = require("./tail");
exports.TailPipe = tail_2.TailPipe;
var truthify_2 = require("./truthify");
exports.TrurthifyPipe = truthify_2.TrurthifyPipe;
var union_2 = require("./union");
exports.UnionPipe = union_2.UnionPipe;
var unique_2 = require("./unique");
exports.UniquePipe = unique_2.UniquePipe;
var without_2 = require("./without");
exports.WithoutPipe = without_2.WithoutPipe;
var pluck_2 = require("./pluck");
exports.PluckPipe = pluck_2.PluckPipe;
var shuffle_2 = require("./shuffle");
exports.ShufflePipe = shuffle_2.ShufflePipe;
var every_2 = require("./every");
exports.EveryPipe = every_2.EveryPipe;
var some_2 = require("./some");
exports.SomePipe = some_2.SomePipe;
var sample_2 = require("./sample");
exports.SamplePipe = sample_2.SamplePipe;
var group_by_2 = require("./group-by");
exports.GroupByPipe = group_by_2.GroupByPipe;
var filter_by_2 = require("./filter-by");
exports.FilterByPipe = filter_by_2.FilterByPipe;
var order_by_2 = require("./order-by");
exports.OrderByPipe = order_by_2.OrderByPipe;
