// ngx-pipes/src/app/pipes/boolean/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var is_defined_1 = require("./is-defined");
var is_null_1 = require("./is-null");
var is_undefined_1 = require("./is-undefined");
var is_string_1 = require("./is-string");
var is_function_1 = require("./is-function");
var is_number_1 = require("./is-number");
var is_array_1 = require("./is-array");
var is_object_1 = require("./is-object");
var is_greater_equal_than_1 = require("./is-greater-equal-than");
var is_greater_than_1 = require("./is-greater-than");
var is_less_equal_than_1 = require("./is-less-equal-than");
var is_equal_to_1 = require("./is-equal-to");
var is_not_equal_to_1 = require("./is-not-equal-to");
var is_identical_to_1 = require("./is-identical-to");
var is_not_identical_to_1 = require("./is-not-identical-to");
var is_less_than_1 = require("./is-less-than");
exports.BOOLEAN_PIPES = [
    is_defined_1.IsDefinedPipe, is_null_1.IsNullPipe, is_undefined_1.IsUndefinedPipe, is_string_1.IsStringPipe, is_function_1.IsFunctionPipe, is_number_1.IsNumberPipe,
    is_array_1.IsArrayPipe, is_object_1.IsObjectPipe, is_greater_equal_than_1.IsGreaterEqualThanPipe, is_greater_than_1.IsGreaterThanPipe, is_less_equal_than_1.IsLessEqualThanPipe,
    is_less_equal_than_1.IsLessEqualThanPipe, is_equal_to_1.IsEqualToPipe, is_not_equal_to_1.IsNotEqualToPipe, is_identical_to_1.IsIdenticalToPipe, is_not_identical_to_1.IsNotIdenticalToPipe,
    is_less_than_1.IsLessThanPipe
];
var NgBooleanPipesModule = (function () {
    function NgBooleanPipesModule() {
    }
    return NgBooleanPipesModule;
}());
NgBooleanPipesModule.decorators = [
    { type: core_1.NgModule, args: [{
                declarations: exports.BOOLEAN_PIPES,
                imports: [],
                exports: exports.BOOLEAN_PIPES
            },] },
];
/** @nocollapse */
NgBooleanPipesModule.ctorParameters = function () { return []; };
exports.NgBooleanPipesModule = NgBooleanPipesModule;
var is_defined_2 = require("./is-defined");
exports.IsDefinedPipe = is_defined_2.IsDefinedPipe;
var is_null_2 = require("./is-null");
exports.IsNullPipe = is_null_2.IsNullPipe;
var is_undefined_2 = require("./is-undefined");
exports.IsUndefinedPipe = is_undefined_2.IsUndefinedPipe;
var is_string_2 = require("./is-string");
exports.IsStringPipe = is_string_2.IsStringPipe;
var is_function_2 = require("./is-function");
exports.IsFunctionPipe = is_function_2.IsFunctionPipe;
var is_number_2 = require("./is-number");
exports.IsNumberPipe = is_number_2.IsNumberPipe;
var is_array_2 = require("./is-array");
exports.IsArrayPipe = is_array_2.IsArrayPipe;
var is_object_2 = require("./is-object");
exports.IsObjectPipe = is_object_2.IsObjectPipe;
var is_greater_equal_than_2 = require("./is-greater-equal-than");
exports.IsGreaterEqualThanPipe = is_greater_equal_than_2.IsGreaterEqualThanPipe;
var is_greater_than_2 = require("./is-greater-than");
exports.IsGreaterThanPipe = is_greater_than_2.IsGreaterThanPipe;
var is_less_equal_than_2 = require("./is-less-equal-than");
exports.IsLessEqualThanPipe = is_less_equal_than_2.IsLessEqualThanPipe;
var is_equal_to_2 = require("./is-equal-to");
exports.IsEqualToPipe = is_equal_to_2.IsEqualToPipe;
var is_not_equal_to_2 = require("./is-not-equal-to");
exports.IsNotEqualToPipe = is_not_equal_to_2.IsNotEqualToPipe;
var is_identical_to_2 = require("./is-identical-to");
exports.IsIdenticalToPipe = is_identical_to_2.IsIdenticalToPipe;
var is_not_identical_to_2 = require("./is-not-identical-to");
exports.IsNotIdenticalToPipe = is_not_identical_to_2.IsNotIdenticalToPipe;
var is_less_than_2 = require("./is-less-than");
exports.IsLessThanPipe = is_less_than_2.IsLessThanPipe;
