// ngx-pipes/src/app/pipes/math/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var max_1 = require("./max");
var min_1 = require("./min");
var percentage_1 = require("./percentage");
var sum_1 = require("./sum");
var floor_1 = require("./floor");
var round_1 = require("./round");
var sqrt_1 = require("./sqrt");
var pow_1 = require("./pow");
var ceil_1 = require("./ceil");
var degrees_1 = require("./degrees");
var bytes_1 = require("./bytes");
var radians_1 = require("./radians");
exports.MATH_PIPES = [
    max_1.MaxPipe, min_1.MinPipe, percentage_1.PercentagePipe, sum_1.SumPipe, floor_1.FloorPipe, round_1.RoundPipe, sqrt_1.SqrtPipe, pow_1.PowerPipe,
    ceil_1.CeilPipe, degrees_1.DegreesPipe, bytes_1.BytesPipe, radians_1.RadiansPipe
];
var NgMathPipesModule = (function () {
    function NgMathPipesModule() {
    }
    return NgMathPipesModule;
}());
NgMathPipesModule.decorators = [
    { type: core_1.NgModule, args: [{
                declarations: exports.MATH_PIPES,
                imports: [],
                exports: exports.MATH_PIPES
            },] },
];
/** @nocollapse */
NgMathPipesModule.ctorParameters = function () { return []; };
exports.NgMathPipesModule = NgMathPipesModule;
var max_2 = require("./max");
exports.MaxPipe = max_2.MaxPipe;
var min_2 = require("./min");
exports.MinPipe = min_2.MinPipe;
var percentage_2 = require("./percentage");
exports.PercentagePipe = percentage_2.PercentagePipe;
var sum_2 = require("./sum");
exports.SumPipe = sum_2.SumPipe;
var floor_2 = require("./floor");
exports.FloorPipe = floor_2.FloorPipe;
var round_2 = require("./round");
exports.RoundPipe = round_2.RoundPipe;
var sqrt_2 = require("./sqrt");
exports.SqrtPipe = sqrt_2.SqrtPipe;
var pow_2 = require("./pow");
exports.PowerPipe = pow_2.PowerPipe;
var ceil_2 = require("./ceil");
exports.CeilPipe = ceil_2.CeilPipe;
var degrees_2 = require("./degrees");
exports.DegreesPipe = degrees_2.DegreesPipe;
var bytes_2 = require("./bytes");
exports.BytesPipe = bytes_2.BytesPipe;
